Stack/Opc.Ua.Core/Schema/SecuredApplicationHelpers.cs

```diff
@@ -371,6 +371,38 @@ public static ServerSecurityPolicyCollection FromListOfSecurityProfiles(ListOfSe
             return policies;
         }
 
+        /// <summary>
+        /// Calculates the security level, given the security mode and policy
+        /// Invalid and none is discouraged
+        /// Just signing is always weaker than any use of encryption
+        /// </summary>
+        public static byte CalculateSecurityLevel(MessageSecurityMode mode, string policyUri)
+        {
+            if ((mode == MessageSecurityMode.Invalid) || (mode == MessageSecurityMode.None))
+            {
+                return 0;
```

Comment:
Now the function is used in a way that could be spoofed, so lets be defensive and return 0 if mode != sign and mode != signandencrypt